pyiceberg/table/update/__init__.py

```diff
@@ -335,19 +330,23 @@ def _(update: SetCurrentSchemaUpdate, base_metadata: TableMetadata, context: _Ta
 
 @_apply_table_update.register(AddPartitionSpecUpdate)
 def _(update: AddPartitionSpecUpdate, base_metadata: TableMetadata, context: _TableMetadataUpdateContext) -> TableMetadata:
+    context.add_update(update)
+    if update.spec.spec_id == INITIAL_PARTITION_SPEC_ID:
+        # no op
+        return base_metadata
```

Comment:
Thanks for the example. 

on a meta level, this is the type of bug I'm afraid of when refactoring... how can we ensure other cases like this are captured

Reply:
I believe adding more tests for this specific case would be helpful. While we have extensive coverage for the logic of updating existing metadata, there are very few tests for `create_table_transaction`, where updates are applied to an empty metadata set to generate the entire metadata. Since this logic is unique to `create_table_transaction`, errors related to it are not detected by the current tests.